src/general.py

```diff
@@ -146,17 +259,32 @@ def generate_sub_object(sub_file: Path, test_name: str) -> htcondor2.Submit:
         job = htcondor2.Submit(f.read())
     if job is None:
         print(f"Error: Invalid submit file for test {test_name}")
-        print("Exiting...")
-        exit(1)
+        sys.exit(1)
 
     # add requirement to land on target ResourceName
+    req_expr = 'TARGET.GLIDEIN_ResourceName == "$(ResourceName)"'
     req = job.get("Requirements")
     if req is None:
-        job["Requirements"] = f'TARGET.GLIDEIN_ResourceName == "$(ResourceName)"'
+        job["Requirements"] = req_expr
     else:
-        job["Requirements"] = (
-            f'TARGET.GLIDEIN_ResourceName == "$(ResourceName)" && ' + req
-        )
+        job["Requirements"] = req_expr + f" && ({req})"
```

Comment:
We could be clever and one line this now that both expressions are stored in variables:
`job["Requirements"] = req_expr if req is None else req_expr + f" && ({req})"`
I am cool with either

Reply:
One line sounds good, I'm not quite fluent in Pythonese, so I forgot you could do this